File: app/src/main/java/com/example/airobserver/onBoarding/SaveState.java
package com.example.airobserver.onBoarding;

import android.content.Context;
import android.content.SharedPreferences;

public class SaveState {
    Context context;
    String saveName;
    SharedPreferences sp;

    public SaveState(Context context, String saveName) {
        this.context = context;
        this.saveName = saveName;
        sp = context.getSharedPreferences(saveName,context.MODE_PRIVATE);
    }

    public void setState(int key)
    {
        SharedPreferences.Editor editor = sp.edit();
        editor.putInt("key",key);
        editor.apply();
    }

    public int getState()
    {
        return sp.getInt("key",0);
    }
}

File: app/src/main/java/com/example/airobserver/ui/onboarding/SaveState.kt
package com.example.airobserver.ui.onboarding

import android.content.Context
import android.content.SharedPreferences

class SaveState(var context: Context, var saveName: String) {
    var sp: SharedPreferences = context.getSharedPreferences(saveName, Context.MODE_PRIVATE)

    var state: Int
        get() = sp.getInt("key", 0)
        set(key) {
            val editor = sp.edit()
            editor.putInt("key", key)
            editor.apply()
        }
}